src/main/java/org/springframework/data/elasticsearch/client/elc/IndicesTemplate.java

```diff
@@ -241,6 +248,45 @@ public Map<String, Object> getMapping() {
 		return responseConverter.indicesGetMapping(getMappingResponse, indexCoordinates);
 	}
 
+	@Override
+	public ClusterMapping getClusterMapping() {
+		GetMappingRequest getMappingRequest = requestConverter.indicesGetMappingRequest(IndexCoordinates.of("*"));
+		GetMappingResponse getMappingResponse = execute(client -> client.getMapping(getMappingRequest));
+
+		return responseConverter.indicesGetMapping(getMappingResponse);
+	}
+
+	/**
+	 * Refreshes the mapping of entities.
+	 * <p>
+	 * This method is responsible for retrieving and updating the metadata related to the entities.
+	 */
+	private void refreshEntitiesMapping() {
+		ClusterMapping clusterMapping = getClusterMapping();
+		for (ClusterMapping.ClusterMappingEntry mappingEntry : clusterMapping) {
+			// Get entity
+			Class<?> entity = null;
+			for (ElasticsearchPersistentEntity<?> persistentEntity : this.elasticsearchConverter.getMappingContext().getPersistentEntities()) {
+				if (mappingEntry.getName().equals(persistentEntity.getIndexCoordinates().getIndexName())) {
```

Comment:
the index name defined on the entity could be different than a simple string, that might be a SpEL expression that needs to be evaluated dynamically

Reply:
Names in **IndexCoordinates** have been resolved

https://github.com/spring-projects/spring-data-elasticsearch/blob/d06c122fd55e55b8886b1d3a3389516b08066784/src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java#L347